testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveViewsCompatibility.java

```diff
@@ -0,0 +1,133 @@
+import io.trino.tempto.ProductTest;
+import io.trino.tempto.assertions.QueryAssert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.ICEBERG;
+import static io.trino.tests.product.TestGroups.STORAGE_FORMATS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestIcebergHiveViewsCompatibility
+        extends ProductTest
+{
+    @Test(groups = {ICEBERG, STORAGE_FORMATS})
+    public void testIcebergHiveViewsCompatibility()
+    {
+        try {
+            // ensure clean env
+            cleanup();
+
+            List<QueryAssert.Row> hivePreexistentTables = onTrino().executeQuery("SHOW TABLES FROM hive.default").rows().stream().map(list -> row(list.toArray())).collect(toList());
+            List<QueryAssert.Row> icebergPreexistentTables = onTrino().executeQuery("SHOW TABLES FROM iceberg.default").rows().stream().map(list -> row(list.toArray())).collect(toList());
+
+            onTrino().executeQuery("CREATE TABLE hive.default.hive_table AS SELECT 1 bee");
+            onTrino().executeQuery("CREATE TABLE iceberg.default.iceberg_table AS SELECT 2 snow");
+
+            onTrino().executeQuery("USE hive.default"); // for sake of unqualified table references
+            onTrino().executeQuery("CREATE VIEW hive.default.hive_view_qualified_hive AS SELECT * FROM hive.default.hive_table");
+            onTrino().executeQuery("CREATE VIEW hive.default.hive_view_unqualified_hive AS SELECT * FROM hive_table");
+            onTrino().executeQuery("CREATE VIEW hive.default.hive_view_qualified_iceberg AS SELECT * FROM iceberg.default.iceberg_table");
+            // this should probably fail but it does not now; testing current behavior as a documentation
+            onTrino().executeQuery("CREATE VIEW hive.default.hive_view_unqualified_iceberg AS SELECT * FROM iceberg_table");
+
+            onTrino().executeQuery("USE iceberg.default"); // for sake of unqualified table references
+            onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_qualified_hive AS SELECT * FROM hive.default.hive_table");
+            assertThatThrownBy(() -> onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_unqualified_hive AS SELECT * FROM hive_table"))
+                    .hasMessageContaining("Not an Iceberg table: default.hive_table");
+            onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_qualified_iceberg AS SELECT * FROM iceberg.default.iceberg_table");
+            onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_unqualified_iceberg AS SELECT * FROM iceberg_table");
+
+            // both hive and iceberg catalogs should list all the views.
```

Comment:
before doing that, consider `USE memory.default` or something

Reply:
WDYM?

Reply:
you use `USE` above to the state isn't "clean". let's `USE` something else to not be biased towards iceberg catalog.

Reply:
Makes sense